handbook/engineering/languages/lang-playbook/terraform.md

```diff
@@ -0,0 +1,17 @@
+# Playbooks for Terraform
+
+## Statefile surgery
+
+_Note: these are provided as internal reference documents. You should rarely need to use this and please reach out to the distribution team or ask in the #dev-ops channel before performing these steps._
+
+To debug some more involved terraform problems you may need to directly modify the state file.
+
+1. Use `TF_DEBUG` environment variable with `WARN`, `DEBUG` logs enabled to locate a possible source of the issue.
+
+1. Use `terraform state pull` to examine the statefile locally.
+
+1. Make the necessary state modifications and perform `terraform state push`.
```

Comment:
I'd like to mention that, if we're only deleting objects, we can just use `terraform state rm`, since that would be less likely to introduce syntax errors.

Reply:
I think in general we should avoid modifying the state by hand as much as possible. Like `state no touchy > state command touchy > state manually touchy`